src/type-checker.cc

```diff
@@ -522,21 +522,14 @@ Result TypeChecker::OnCallIndirect(const TypeVector& param_types,
 
 Result TypeChecker::OnIndexedFuncRef(Index* out_index) {
   Type type;
-  CHECK_RESULT(PeekType(0, &type));
-  Result result = Result::Ok;
-  if (!(type == Type::Any || type.IsReferenceWithIndex())) {
-    TypeVector actual;
-    actual.push_back(type);
-    std::string message =
-        "type mismatch in call_ref, expected reference but got " +
-        TypesToString(actual);
-    PrintError("%s", message.c_str());
-    result = Result::Error;
+  Result result = PeekType(0, &type);
+  if (!type.IsReferenceWithIndex()) {
```

Comment:
Is it OK to use `type` here even if PeekType fails?  i.e. should we return early if PeekType fails?

Reply:
yeah this is fine. PeekType will set Type to Any